### PR TITLE
LeakyReLU implementation

### PR DESCRIPTION
**Pull request by rotmanmi**
in case you choose a negative slope, it automatically allocates more memory.

### CODE REVIEW

**Comment by szagoruyko** [comment]
can you give a reason to have negative slope in leaky relu?

**Comment by rotmanmi** [comment]
http://arxiv.org/abs/1412.6830

This is an extension to Leaky ReLU, but they do have negative slopes there.

**Comment by szagoruyko** [comment]
you can add inplace option to prelu and instantiate leaky relu from it, I am against having redundant code in nn. Negative slope would also work with it, even though it's importance is arguable.

**Comment by soumith** [comment]
Michael, I am ready to merge it, if you squash your commits, add docs and unit tests.

File: generic/LeakyReLU.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  THTensor *input = luaT_checkudata(L, 2, torch_Tensor);
+  THTensor *gradOutput = luaT_checkudata(L, 3, torch_Tensor);
+  real negval = luaT_getfieldchecknumber(L, 1, "negval");
+  THTensor *gradInput = luaT_getfieldcheckudata(L, 1, "gradInput", torch_Tensor);
+  int inPlace = luaT_getfieldcheckboolean(L, 1, "inplace");
+
+  if (inPlace) {
+    TH_TENSOR_APPLY2(real, gradOutput, real, input,    \
+                     if ((*input_data) <= 0) { \
+                       *gradOutput_data = negval;           \
```
Comment (andreaskoepf) [inline generic/LeakyReLU.c]: This should be a multiplication with negval instead a direct assignment (e.g. ``` *gradOutput_data *= negval; ``` )... 

Same is true for ln. 45 which should become
``` 
*gradInput_data = (*input_data > 0) ? *gradOutput_data :  *gradOutput_data * negval; 
```

**Comment by soumith** [comment]
already merged in another PR.

**Pull request closed without merging**